osf/management/commands/migrate_pagecounter_data.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import connection
+
+from framework import sentry
+
+logger = logging.getLogger(__name__)
+
+
+LIMIT_CLAUSE = ' LIMIT %s);'
+NO_LIMIT_CLAUSE = ');'
+
+REVERSE_SQL_BASE = '''
+UPDATE osf_pagecounter PC
+SET
+    resource_id = NULL,
+    file_id = NULL,
+    version = NULL
+WHERE PC.id in (
+    select PC.id from osf_pagecounter PC
+        left outer join osf_guid Guid on Guid._id = split_part(PC._id, ':', 2)
+        left outer join osf_basefilenode File on File._id = split_part(PC._id, ':', 3)
+    where
+          resource_id IS NOT NULL OR
+          file_id IS NOT NULL AND
+          Guid._id IS NOT NULL AND File._id IS NOT NULL
+'''
+REVERSE_SQL = '{} {}'.format(REVERSE_SQL_BASE, NO_LIMIT_CLAUSE)
+REVERSE_SQL_LIMITED = '{} {}'.format(REVERSE_SQL_BASE, LIMIT_CLAUSE)
+
+FORWARD_SQL_BASE = '''
+    UPDATE osf_pagecounter PC
+    SET
+        action = split_part(PC._id, ':', 1),
+        resource_id = Guid.id,
+        file_id = File.id,
+        version = NULLIF(split_part(PC._id, ':', 4), '')::int
+    FROM osf_guid Guid, osf_basefilenode File
+        WHERE
+              Guid._id = split_part(PC._id, ':', 2) AND
+              File._id = split_part(PC._id, ':', 3) AND
+              PC.id in (
+                  select PC.id from osf_pagecounter PC
+                      left outer join osf_guid Guid on Guid._id = split_part(PC._id, ':', 2)
+                      left outer join osf_basefilenode File on File._id = split_part(PC._id, ':', 3)
```

Comment:
Yes, it was the best reason: because it completely didn't occur to me to use an inner join. :)

Reply:
Oh, no, that was all lies. I could have used the INNER JOIN on the count query below and probably in the subquery, but I can't use the implicit join approach here because I can't limit an update query without SELECT. 

But the INNER JOIN on the subquery looks like it works great, so I'll fix that on all the places.

Reply:
ah I see!